service/internal/nodecount/node_count.go

```diff
@@ -132,14 +138,13 @@ func (nc *NodeCount) cachedNodes(ctx context.Context, cr metav1.Object) (corev1.
 	return nodes, nil
 }
 
-func (nc *NodeCount) lookupNodes(ctx context.Context) (corev1.NodeList, error) {
-	// TODO we need to get rid off the controllercontext.Context but for now it should be fine.
-	cc, err := controllercontext.FromContext(ctx)
+func (nc *NodeCount) lookupNodes(ctx context.Context, cr metav1.Object) (corev1.NodeList, error) {
+	client, err := nc.tenantClient.K8sClient(ctx, cr)
 	if err != nil {
```

Comment:
This here needs graceful error handling in case the tenant client could not be provided.